File: FrontEnd/src/modules/Trader/models/SchedulePlanInfoView.ts
import type { OrderMode } from "./SchedulePlanCommand";

export interface ScheduledOrderInfoView {
  id: string;
  accountId: string;
  accountName?: string | null;
  symbolIsin: string;
  symbolName?: string | null;
  side: number;
  mode: OrderMode;
  quantity: string;
  totalValue: string;
  time: string;
  /** آیا این سفارش توسط بک‌اند ارسال شده */
  fired: boolean;
}

export type SchedulePlanStatus =
  | "idle"
  | "waiting"
  | "logging-in"
  | "refreshing"
  | "waiting-fire"
  | "firing"
  | "done"
  | "cancelled"
  | "error";

export interface SchedulePlanInfoView {
  id: string;
  name: string;
  date: string;
  enabled: boolean;
  autoLoginAt: string;
  autoRefreshAt: string;
  orders: ScheduledOrderInfoView[];
  status: SchedulePlanStatus;
  message?: string | null;
}

File: Trader.Application/Dtos/SchedulePlanInfoView.cs
namespace Trader.Application.Dtos
{
    public class SchedulePlanInfoView
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = default!;
        public string Date { get; set; } = default!;         // "YYYY-MM-DD"
        public bool Enabled { get; set; }
        public string AutoLoginAt { get; set; } = default!;  // "HH:MM:SS"
        public string AutoRefreshAt { get; set; } = default!;

        public List<ScheduledOrderInfoView> Orders { get; set; } = new();

        public string Status { get; set; } = "idle";
        public string? Message { get; set; }
    }
}